pkg/webhook/admission/validator.go

```diff
@@ -54,29 +69,27 @@ func (h *validatingHandler) Handle(ctx context.Context, req Request) Response {
 	if h.validator == nil {
 		panic("validator should never be nil")
 	}
-
 	// Get the object in the request
 	obj := h.validator.DeepCopyObject().(Validator)
-	if req.Operation == v1.Create {
-		err := h.decoder.Decode(req, obj)
-		if err != nil {
-			return Errored(http.StatusBadRequest, err)
-		}
 
-		err = obj.ValidateCreate()
+	var err error
+	var warnings []string
+
+	switch req.Operation {
+	case v1.Connect:
+		// No validation for connect requests.
+		// TODO(vincepri): Should we validate CONNECT requests? In what cases?
+	case v1.Create:
+		err = h.decoder.Decode(req, obj)
 		if err != nil {
```

Comment:
```suggestion
		if err := h.decoder.Decode(req, obj); err != nil {
```
nit: just feels a bit cleaner

(also reduces the diff between the 2 validators)

Same for the other DecodeRaw calls below